language/tools/transaction-replay/src/transaction_debugger_interface.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{bail, Result};
+use libra_state_view::StateView;
+use libra_types::{
+    access_path::AccessPath,
+    account_address::AccountAddress,
+    account_state::AccountState,
+    account_state_blob::AccountStateBlob,
+    transaction::{Transaction, Version},
+};
+use libradb::LibraDB;
+use std::{convert::TryFrom, path::Path, sync::Arc};
+use storage_interface::DbReader;
+
+pub trait StorageDebuggerInterface {
+    fn get_account_state_by_version(
```

Comment:
I defined the API needed to be provided from the system here. Currently it is implemented with a read-only Local DB instance. In the future we would like this to be done via some rpc services to be able to run this tool outside of production environment.

Reply:
Even if json-rpc or a similar service doesn't support fetching the state at *every* version (which would be a lot of data!), perhaps someone could provide a lighter API for fetching the state at (e.g.) every epoch change or every waypoint?